### ISSUE TITLE
Badge images in README are broken

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#1056]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/1056)
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 25, 2019, 22:28

## Summary

This is a relatively very minor inconvenience, but a couple of broken images at the very start of our README isn't ideal either :)

## Steps to reproduce

View README at https://gitlab.com/BuildStream/buildstream/blob/master/README.rst.

## What is the current bug behavior?

Image sources for the smapshot and release badges point to URLs that return 404.

## Possible fixes

One simple fix would be to revert !876. Seems like the "old" links are still correct. However, I personally don't quite understand the split between the two websites so wanted to get a second opinion.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 25, 2019, 22:29

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 25, 2019, 22:29

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jun 26, 2019, 10:52

marked this issue as related to #178

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jun 26, 2019, 11:00

Ok so let's explain what's going on here, and first point out that the root cause of all of this madness is essentially still unresolved #178 

* Because of #178, we have docs published in more than one place
* Docs in https://buildstream.gitlab.io/buildstream/ are the new BuildStream 2 API
* Docs in https://docs.buildstream.build/ are the currently released 1.2 docs
* Because of *"technical difficulties"*, the docs on https://docs.buildstream.build/ are published using a process managed by https://gitlab.com/BuildStream/docs-website
  * This essentially downloads the last built docs for `bst-1.2` from the the gitlab artifacts and publishes it to a space which is reached by https://docs.buildstream.build/
* The badges are generated in BuildStream master, these are generated by way of introspecting the latest release tags in the git repo whenever docs get built

In summary, the correct place to obtain the badges should always be `master`, regardless of current release version or what documentation they appear in (the whole point of the badges is to reflect what is the latest dev and latest stable tag, as such even documentation published last year contains a reference to the latest badges built at https://buildstream.gitlab.io/buildstream/).

I *suspect* that the issue here is that the README is referring to a https://docs.buildstream.build/ based url, which of course does not have the badges as the badge generation was never added in `bst-1.2` branch.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 26, 2019, 13:50

mentioned in commit fdb89962201e294b820bf897a5de074e9aca0464

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 26, 2019, 13:51

mentioned in merge request !1422

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 26, 2019, 13:53

[[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) thanks very much for the explanation.

Currently the badges are indeed pointing to docs.buildstream.build, which throws a 404. I have submitted a fix in !1422.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jun 26, 2019, 13:55

mentioned in commit 1a103e09763754a32ca287707e212e4842fa6587

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Jun 26, 2019, 15:01

closed via merge request !1422

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @marge-bot123]](https://gitlab.com/marge-bot123) on Jun 26, 2019, 15:01

mentioned in commit 7224582503f85af5e5bcf072c2d1ae8d9ce2c19e

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @cs-shadow]](https://gitlab.com/cs-shadow) on Jul 1, 2019, 17:38

mentioned in commit c3e2be3b092f63951d283be2738f878dcbb24b26
